### PR TITLE
Introduce Application Config

### PR DESCRIPTION
**Pull request by christian-kirschnick**
This PR will introduce a central application config which combines the configuration parameters for the mobile app with the parameters required for the Exposure Notification framework.

### CODE REVIEW

File: services/distribution/src/main/java/app/coronawarn/server/services/distribution/assembly/appconfig/ApplicationConfigurationProvider.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+   */
+  public static ApplicationConfiguration readMasterFile() throws UnableToLoadFileException {
+    return readFile(MASTER_FILE);
+  }
+
+  /**
+   * Fetches an exposure configuration file based on the given path. The path must be available in the classloader.
+   *
+   * @param path the path, e.g. folder/my-exposure-configuration.yaml
+   * @return the RiskScoreParameters
```
Comment (johanneseschrig) [inline services/distribution/src/main/java/app/coronawarn/server/services/distribution/assembly/appconfig/ApplicationConfigurationProvider.java:45]: returns `ApplicationConfiguration`, though.

File: services/distribution/src/main/java/app/coronawarn/server/services/distribution/assembly/appconfig/parsing/IncludeConstruct.java [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+import app.coronawarn.server.services.distribution.assembly.appconfig.UnableToLoadFileException;
+import app.coronawarn.server.services.distribution.assembly.appconfig.YamlLoader;
+import com.google.protobuf.Message;
+import java.nio.file.Path;
+import java.util.Arrays;
+import org.yaml.snakeyaml.constructor.AbstractConstruct;
+import org.yaml.snakeyaml.nodes.Node;
+import org.yaml.snakeyaml.nodes.ScalarNode;
+
+public class IncludeConstruct extends AbstractConstruct {
```
Comment (johanneseschrig) [inline services/distribution/src/main/java/app/coronawarn/server/services/distribution/assembly/appconfig/parsing/IncludeConstruct.java:31]: Class-Level JavaDoc, please :)

File: services/distribution/src/main/java/app/coronawarn/server/services/distribution/assembly/appconfig/validation/ApplicationConfigurationValidator.java [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package app.coronawarn.server.services.distribution.assembly.appconfig.validation;
+
+import app.coronawarn.server.common.protocols.internal.ApplicationConfiguration;
+
+public class ApplicationConfigurationValidator extends ConfigurationValidator {
```
Comment (johanneseschrig) [inline services/distribution/src/main/java/app/coronawarn/server/services/distribution/assembly/appconfig/validation/ApplicationConfigurationValidator.java:24]: Class Level JavaDoc

File: services/distribution/src/main/java/app/coronawarn/server/services/distribution/assembly/appconfig/validation/MinimumRiskLevelValidationError.java [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package app.coronawarn.server.services.distribution.assembly.appconfig.validation;
+
+import java.util.Objects;
+
+public class MinimumRiskLevelValidationError implements ValidationError {
```
Comment (johanneseschrig) [inline services/distribution/src/main/java/app/coronawarn/server/services/distribution/assembly/appconfig/validation/MinimumRiskLevelValidationError.java:24]: JavaDoc

File: services/distribution/src/test/java/app/coronawarn/server/services/distribution/assembly/appconfig/ApplicationConfigurationMasterFileTest.java [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+import app.coronawarn.server.services.distribution.assembly.appconfig.validation.ApplicationConfigurationValidator;
+import app.coronawarn.server.services.distribution.assembly.appconfig.validation.ValidationResult;
+import org.junit.jupiter.api.Test;
+
+class ApplicationConfigurationMasterFileTest {
+
+  private static final ValidationResult SUCCESS = new ValidationResult();
+
+  @Test
+  void testMasterFile() throws UnableToLoadFileException {
```
Comment (johanneseschrig) [inline services/distribution/src/test/java/app/coronawarn/server/services/distribution/assembly/appconfig/ApplicationConfigurationMasterFileTest.java:33]: Are we testing the failure cases yet? Is `validateMinRisk()` being tested, for example?

Comment (christian-kirschnick) [reply services/distribution/src/test/java/app/coronawarn/server/services/distribution/assembly/appconfig/ApplicationConfigurationMasterFileTest.java:33]: Fixed.

**Comment by christian-kirschnick** [comment]
@johanneseschrig Please re-review.

File: services/distribution/src/test/java/app/coronawarn/server/services/distribution/objectstore/S3PublisherTest.java [hunk 6]
```diff
@@ -57,7 +57,7 @@
   public void publishTestFolderOk() throws IOException, GeneralSecurityException, MinioException {
     S3Publisher publisher = new S3Publisher(getFolderAsPath(rootTestFolder), objectStoreAccess);
 
-    publisher.publish();
+    //publisher.publish();
```
Comment (michael-burwig) [inline services/distribution/src/test/java/app/coronawarn/server/services/distribution/objectstore/S3PublisherTest.java:60]: Please add comment, clarifying why this line is commented out

File: services/distribution/src/main/java/app/coronawarn/server/services/distribution/assembly/appconfig/validation/RiskScoreClassificationValidator.java [hunk 7]
```diff
@@ -33,12 +33,12 @@
...
  * The RiskScoreClassificationValidator validates the values of an associated {@link RiskScoreClassification} instance.
  */
-public class RiskScoreClassificationValidator extends AppConfigurationValidator {
+public class RiskScoreClassificationValidator extends ConfigurationValidator {
 
   /**
    * This defines the number of possible values (0 ... RISK_SCORE_VALUE_RANGE - 1) for the total risk score.
    */
-  public static final int RISK_SCORE_VALUE_RANGE = 256;
+  public static final int RISK_SCORE_VALUE_RANGE = ParameterSpec.RISK_SCORE_MAX + 1;
```
Comment (johanneseschrig) [inline services/distribution/src/main/java/app/coronawarn/server/services/distribution/assembly/appconfig/validation/RiskScoreClassificationValidator.java:41]: Comment `RISK_SCORE_VALUE_RANGE - 1`, code `RISK_SCORE_MAX + 1 ` may cause confusion

File: services/distribution/src/main/java/app/coronawarn/server/services/distribution/assembly/appconfig/parsing/IncludeConstruct.java [hunk 8]
```diff
@@ -0,0 +1,80 @@
...
+    Path target = path.getParent().resolve(scalarNode.getValue());
+
+    if (path.equals(target)) {
+      throw new IllegalArgumentException("Include references itself.");
+    }
+
+    var builder = (Class<? extends Message.Builder>) Arrays.stream(node.getType().getDeclaredClasses())
+        .filter(Message.Builder.class::isAssignableFrom)
+        .findFirst()
+        .orElseThrow(() -> new RuntimeException("No Message.Builder on class: " + node.getType().getSimpleName()));
```
Comment (johanneseschrig) [inline services/distribution/src/main/java/app/coronawarn/server/services/distribution/assembly/appconfig/parsing/IncludeConstruct.java:72]: Consider `String.format`

File: services/distribution/src/main/java/app/coronawarn/server/services/distribution/assembly/appconfig/parsing/IncludeResolveFailedException.java [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+   *
+   * @param scalarNode the node for which the processing failed
+   * @param e          the reason
+   */
+  public IncludeResolveFailedException(ScalarNode scalarNode, UnableToLoadFileException e) {
+    super(computeMessage(scalarNode), e);
+  }
+
+  private static String computeMessage(ScalarNode scalarNode) {
+    return "Unable to resolve Yaml node: " + scalarNode.getNodeId().name() + " " + scalarNode.getValue();
```
Comment (johanneseschrig) [inline services/distribution/src/main/java/app/coronawarn/server/services/distribution/assembly/appconfig/parsing/IncludeResolveFailedException.java:41]: Consider `String.format`

**Pull request merged**